hub/core/chunk_engine.py

```diff
@@ -1291,11 +1291,18 @@ def _get_full_chunk(self, index):
         threshold = 10
         if type(index.values[0].value) == slice:
             start = index.values[0].value.start or 0
-            stop = index.values[0].value.stop
+            stop = index.values[0].value.stop or self.tensor_meta.length
```

Comment:
use self.num_samples and also update self.num_samples of chunk_engine to use same logic as tensor.num_samples (i.e. account for sequences)

Reply:
I understood first part of your comment, but didn't fully get the second part. How should I update the self.num_samples?

Reply:
Look at hub/core/tensor.py's num_samples method